### ISSUE TITLE
Neoverse-V2 detected as neoverse_n1 on Nvidia Grace

### ISSUE THREAD

**Issue by Candles** [LLNL/spack#43106]
### Steps to reproduce

On an Nvidia Grace Hopper server, the target architecture per `spack arch` is detected as neoverse_n1 when it should be detected as a neoverse_v2.

### Error message

Current output:
```
[olsont@youmu ~]$ spack arch
linux-rocky9-neoverse_n1
```
Expected output:
```
[olsont@youmu ~]$ spack arch
linux-rocky9-neoverse_v2
```

### Information on your system

```
[olsont@youmu ~]$ lscpu
Architecture:           aarch64
  CPU op-mode(s):       64-bit
  Byte Order:           Little Endian
CPU(s):                 72
  On-line CPU(s) list:  0-71
Vendor ID:              ARM
  Model name:           Neoverse-V2
    Model:              0
    Thread(s) per core: 1
    Core(s) per socket: 72
    Socket(s):          1
    Stepping:           r0p0
    Frequency boost:    disabled
    CPU max MHz:        3438.0000
    CPU min MHz:        81.0000
    BogoMIPS:           2000.00
    Flags:              fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 sm3 sm4 asimddp sha512 sve asimdfhm dit uscat ilrcpc flagm ssbs sb dcpodp sve2 sveaes svepmull svebitperm svesha3 svesm4 flagm2 frint svei8mm svebf16 i8mm bf16 dgh

[olsont@youmu ~]$ head /proc/cpuinfo
processor       : 0
BogoMIPS        : 2000.00
Features        : fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 sm3 sm4 asimddp sha512 sve asimdfhm dit uscat ilrcpc flagm ssbs sb dcpodp sve2 sveaes svepmull svebitperm svesha3 svesm4 flagm2 frint svei8mm svebf16 i8mm bf16 dgh
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd4f
CPU revision    : 0

[olsont@yuyuko ~]$ spack debug report
* **Spack:** 0.21.2 (89319413d5d90caa9bea98e3419f8aeab61a2a0d)
* **Python:** 3.9.18
* **Platform:** linux-rocky9-neoverse_n1
* **Concretizer:** clingo
```

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

**Comment by aweits** [comment]
a quick note (and tagging @alalazo), because I've run into this and done the homework:

RHEL-ish kernels have kernel configs like so:
```
# CONFIG_ARM64_BTI is not set
# CONFIG_ARM64_PTR_AUTH is not set
```

which suppresses paca, pacg, and bti in flags. This needs fixing in archspec.

a workaround is to remove those flags from the entry in lib/spack/external/archspec/json/cpu/microarchitectures.json

**Comment by alalazo** [comment]
See https://github.com/archspec/archspec-json/issues/83

**Comment by alalazo** [comment]
> a workaround is to remove those flags from the entry in lib/spack/external/archspec/json/cpu/microarchitectures.json

As of https://github.com/spack/spack/pull/42854 people can also have a JSON file somewhere, overriding just `neoverse_v2` and setting `ARCHSPEC_EXTENSION_CPU_DIR` to the directory containing it. That said, I'll try to get to the issue above shortly.

**Comment by alalazo** [comment]
Should be fixed in v0.22 by #44022
